test/colony.js

```diff
@@ -1720,4 +1721,88 @@ contract("Colony", addresses => {
+      await colony.setRecoveryRole(addresses[2], { from: owner });
+
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 2);
+
+      await colony.removeRecoveryRole(addresses[2], { from: owner });
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 1);
+
+      // Can't remove twice
+      await colony.removeRecoveryRole(addresses[2], { from: owner });
+      numRecoveryRoles = await colony.numRecoveryRoles();
+      assert.equal(numRecoveryRoles.toNumber(), 1);
+    });
+
+    it("should not exit recovery mode without approvals", async () => {
+      const owner = addresses[0];
+      await colony.setRecoveryRole(addresses[1], { from: owner });
+      await colony.setRecoveryRole(addresses[2], { from: owner });
+
+      await colony.enterRecoveryMode({ from: owner });
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      await checkErrorRevert(colony.exitRecoveryMode());
+
+      // 1/3 approve
+      await colony.approveExitRecovery({ from: owner });
+      await checkErrorRevert(colony.exitRecoveryMode());
+    });
+
+    it("should exit recovery mode with sufficient approvals", async () => {
+      const owner = addresses[0];
+      await colony.setRecoveryRole(addresses[1], { from: owner });
+      await colony.setRecoveryRole(addresses[2], { from: owner });
+
+      await colony.enterRecoveryMode({ from: owner });
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      // 2/3 approve
+      await colony.approveExitRecovery({ from: owner });
+      await colony.approveExitRecovery({ from: addresses[1] });
+      await colony.exitRecoveryMode();
+    });
+
+    it("users cannot approve twice", async () => {
+      const owner = addresses[0];
+
+      await colony.enterRecoveryMode({ from: owner });
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      await colony.approveExitRecovery({ from: owner });
+      await checkErrorRevert(colony.approveExitRecovery({ from: owner }));
+    });
+
+    it("should allow editing of general variables", async () => {
+      const owner = addresses[0];
+
+      await colony.enterRecoveryMode({ from: owner });
+
+      const unprotected0 = await web3GetStorageAt(colony.address, 5);
+      await colony.setStorageSlotRecovery(5, fromAscii("0xdeadbeef"), { from: owner });
+
+      const unprotected1 = await web3GetStorageAt(colony.address, 5);
+      assert.notEqual(unprotected0.toString(), unprotected1.toString());
+    });
+
+    it("should not allow editing of protected variables", async () => {
+      const owner = addresses[0];
+      const protectedLoc = 0;
+
+      await colony.enterRecoveryMode({ from: owner });
+      await checkErrorRevert(colony.setStorageSlotRecovery(protectedLoc, fromAscii("0xdeadbeef"), { from: owner }));
```

Comment:
Strictly, it doesn't need correcting, but it's weird you're using `fromAscii` on `0xdeadbeef`, which means it stores the bytes for the ASCII representation of the string `0xdeadbeef` rather than the bytes `0xdeadbeef`.

Reply:
That's fair. I was using `fromAscii` before to pass in `bytes32` and just kept using it. I can make it another string so it's not confusing.